src/Xamarin.Android.Build.Tasks/Tasks/JavaDoc.cs

```diff
@@ -53,9 +62,27 @@ protected override string GenerateCommandLineCommands ()
 			cmd.AppendFileNameIfNotNull (context_src);
 			cmd.AppendSwitch ("-subpackages");
 			cmd.AppendSwitch (".");
+			var cps = ReferenceJars?.ToList () ?? new List<string> ();
+			if (JavaPlatformJar != null)
+				cps.Add (JavaPlatformJar);
+			if (cps.Any ()) {
+				if (Environment.OSVersion.Platform != PlatformID.Unix)
+					cmd.AppendSwitch ("-cp " + string.Join (";", cps.Select (cp => '"' + cp + '"')));
+				else 
+					cmd.AppendSwitch ("-cp " + string.Join (":", cps));
```

Comment:
Unix *also* supports directory-names-containing-spaces. (Yes, perhaps that's not "advisable," but they can still exist.) As such, `cps` really should be quoted as well.

Reply:
Did you actually try to quote paths? To my experience that results in directory lookup for paths that contain quotations chars and results in error.

Reply:
Yes, and it works for me. Consider: [cp.zip](https://github.com/xamarin/xamarin-android/files/2024054/cp.zip)

```sh
$ curl -o cp.zip https://github.com/xamarin/xamarin-android/files/2024054/cp.zip
$ unzip cp.zip
$ cd cp
$ make all run
mkdir bin
javac -d bin Example.java
mkdir -p bin\ 1/foo/bar
mv bin/foo/bar/B.class bin\ 1/foo/bar
mkdir -p bin\ 2/foo/bar
mv bin/foo/bar/C.class bin\ 2/foo/bar
java -cp "bin:bin 1:bin 2" foo.bar.App
```

Here we have a `java -cp` invocation which quotes paths with spaces. It works.